App/Program.fs

```diff
@@ -22,46 +23,50 @@ let writeUrl =
 let readUrl =
   Environment.tryGetEnv "MESSAGE_DB_REPLICA_URL" |> Option.defaultValue writeUrl
 
-let connection = MessageDbConnector(writeUrl, readUrl).Establish()
-let context = MessageDbContext(connection)
+let connection = MessageDbClient(writeUrl, readUrl)
+let context = MessageDbContext(connection, 100)
 let caching = CachingStrategy.SlidingWindow(cache, TimeSpan.FromMinutes(20))
 
 let service =
   MessageDbCategory(context, Invoice.Events.codec, Invoice.Fold.fold, Invoice.Fold.initial, caching)
   |> Equinox.Decider.resolve log
   |> Invoice.create
 
-let builder = WebApplication.CreateBuilder()
-let app = builder.Build()
-
-let raiseInvoice body =
-  task {
-    let id = Guid.NewGuid() |> Invoice.InvoiceId.ofGuid
-    do! service.Raise(id, body)
-    return id
-  }
-
-app.MapPost("/", Func<_, _>(raiseInvoice)) |> ignore
-
-let finalizeInvoice id =
-  task {
-    do! service.Finalize(id)
-    return "OK"
-  }
-
-app.MapPost("/{id}/finalize", Func<_, _>(finalizeInvoice)) |> ignore
+let numberService =
+  let access = AccessStrategy.AdjacentSnapshots(InvoiceNumbering.Fold.snapshotEventType, InvoiceNumbering.Fold.toSnapshot)
+  MessageDbCategory(context, InvoiceNumbering.Events.codec, InvoiceNumbering.Fold.fold, InvoiceNumbering.Fold.initial, caching, access)
+  |> Equinox.Decider.resolve log
+  |> InvoiceNumbering.create
 
-let recordPayment id payment =
-  task {
-    do! service.RecordPayment(id, payment)
-    return "OK"
-  }
+let reactorService = InvoiceNumberingReactor.Service(numberService, service)
```

Comment:
arguably this could become `InvoiceNumberingReactor.create (context, cache)`, and then have that create the two relevant Domain Services 

(that then also helps with exposing such helpers for test scenarios)

Reply:
see `type Domain.Config.Store` in proHotel
My convention is to always have wiring for memory and >= 1 concrete store as a DU
Then you can look at top of an aggregate module to see where it lives and bottom of it to see the access strategy

The obvious debatable bit is that this means Domain.fsproj has a PackageRef to MemoryStore and ConcreteStore

This is also slightly coloured by the fact that dotnet-templates is in the business of providing store switching

One fringe benefit is that this means one can mix and match and/or migrate a given Category to another store easily by adding a DU case and fixing the errors (which I did, from Cosmos to Dynamo, one time!).

I do agree that stuff gets noisier if you consider more than one store target, especially for a simple sample.

Over to you to do as you see fit!